### ISSUE TITLE
renaming jellyfish to jellyfish1 and tagging 1.1.12 release

### ISSUE THREAD

**Issue by mmokrejs** [gmarcais/Jellyfish#121]
Hi,
  it is very inconvenient that jellyfish-1 and jellyfish install insto same paths and both provide /usr/bin/jellyfish. A number of tools relies on jellyfish-1 while other on jellyfish-2. They cannot coexist due to this.

  I propose renaming the jellyfish (version 1) binary to jellyfish1 and likewise the pkg-config file and header contain jellyfish1 instead.

  Below is what works for me, but does not really create a patch. But I guess you get my point:

```
        ED=${DESTDIR}
        PN=jellyfish
        SLOT=1
        get_libdir="lib64"
        mv "${ED}"/usr/bin/jellyfish "${ED}"/usr/bin/jellyfish1 || die
        sed -e "s#jellyfish-${PV}#jellyfish1#" -i "${ED}/usr/$(get_libdir)"/pkgconfig/jellyfish-1.1.pc || die
        find "${ED}"/usr/include/"${P}"/"${PN}" -type f | while read f; do
                sed -e "s#include <jellyfish/#include <jellyfish1/#" -i $f || die
        done
        mkdir -p "${ED}/usr/include/${PN}${SLOT}" || die
        mv "${ED}"/usr/include/"${P}"/"${PN}"/* "${ED}/usr/include/${PN}${SLOT}/" || die
        rm -r "${ED}/usr/include/${P}" || die
        mv "${ED}"/usr/share/man/man1/jellyfish.1.bz2 "${ED}"/usr/share/man/man1/jellyfish"${SLOT}".1.bz2 || die
```

Of course, likewise I propose installing jellyfish-2 into /usr/include/jellyfish2 , the binary as /usr/bin/jellyfish2, pkg-config file adjusted accordingly.

BTW, Debian and Ubuntu do this already, and I started to do same for Gentoo.

**Comment by mmokrejs** [comment]
> I don't quite understand that comment. The headers are installed in their own subdirectories by default and should not clash

Both `jellyfish-1.x` and `jellyfish-2.x` packages install into `/usr/include/jellyfish/` so they do clash. Also, both provide `/usr/bin/jellyfish`. I did not know about `./configure --program-suffix=-2`, thank you but I assume that does not fix the include directory naming. And I would like to stick to the `jellyfish{1,2}` style which is I think already present in Debian.

My intent (and the current Gentoo patch) is to be able to have both version installed concurrently (meaning headers, libs, binaries).

The patches for the 3rd-party tools I was aware of are already existing in the URL I posted initially. It would help a lot to cleanup the mess.

**Comment by gmarcais** [comment]
I don't know what Gentoo or Debian do with their packages, but jellyfish 1 and 2 can be installed without difficulty side-by-side. The headers are installed in their own sub-directory and are properly versioned. The libraries do not clash either. The autoconf/automake distribution packages are correct in that respect (see example below). I am all in favor for the Gentoo packages to install side by side, but I believe it is already supported from the source tar ball. Let me know if you need an extra feature not yet supported.

Here is a quick example. The output of the command is removed, excepted for the last `ls`. As you can see, there is no clash of files, even though the two versions are installed in the same directory called `inst`.
```Shell
$ tar zxf ../series1/build-gcc/default/jellyfish-1.1.11.tar.gz
$ tar zxf ../build-gcc/default/jellyfish-2.2.10.tar.gz
$ (cd jellyfish-1.1.11; ./configure --prefix=$(pwd)/../inst --program-suffix=-1 && make -j 4 install) >/dev/null
$ (cd jellyfish-2.2.10; ./configure --prefix=$(pwd)/../inst --program-suffix=-2 && make -j 4 install) >/dev/null
$ ls -lR inst
inst:
total 16
drwxr-xr-x 2 gus users 4096 Apr 30 11:24 bin
drwxr-xr-x 4 gus users 4096 Apr 30 11:24 include
drwxr-xr-x 3 gus users 4096 Apr 30 11:24 lib
drwxr-xr-x 3 gus users 4096 Apr 30 11:23 share

inst/bin:
total 7224
-rwxr-xr-x 1 gus users 2868952 Apr 30 11:23 jellyfish-1
-rwxr-xr-x 1 gus users 4523392 Apr 30 11:24 jellyfish-2

inst/include:
total 8
drwxr-xr-x 3 gus users 4096 Apr 30 11:23 jellyfish-1.1.11
drwxr-xr-x 3 gus users 4096 Apr 30 11:24 jellyfish-2.2.10

inst/include/jellyfish-1.1.11:
total 4
drwxr-xr-x 2 gus users 4096 Apr 30 11:23 jellyfish

inst/include/jellyfish-1.1.11/jellyfish:
total 1344
-rw-r--r-- 1 gus users   5028 Apr 30 11:23 aligned_values_array.hpp
-rw-r--r-- 1 gus users   1617 Apr 30 11:23 allocators_malloc.hpp
-rw-r--r-- 1 gus users   1817 Apr 30 11:23 allocators_mmap.hpp
-rw-r--r-- 1 gus users   2637 Apr 30 11:23 allocators_shm.hpp
-rw-r--r-- 1 gus users   1846 Apr 30 11:23 atomic_gcc.hpp
-rw-r--r-- 1 gus users    731 Apr 30 11:23 backtrace.hpp
-rw-r--r-- 1 gus users   1958 Apr 30 11:23 capped_integer.hpp
-rw-r--r-- 1 gus users   1767 Apr 30 11:23 circular_buffer.hpp
-rw-r--r-- 1 gus users   6009 Apr 30 11:23 compacted_dumper.hpp
-rw-r--r-- 1 gus users  19097 Apr 30 11:23 compacted_hash.hpp
-rw-r--r-- 1 gus users   4973 Apr 30 11:23 concurrent_queues.hpp
-rw-r--r-- 1 gus users   1453 Apr 30 11:23 counter.hpp
-rw-r--r-- 1 gus users   3999 Apr 30 11:23 dbg.hpp
-rw-r--r-- 1 gus users   4702 Apr 30 11:23 direct_indexing_array.hpp
-rw-r--r-- 1 gus users   4695 Apr 30 11:23 direct_sorted_dumper.hpp
-rw-r--r-- 1 gus users   3080 Apr 30 11:23 divisor.hpp
-rw-r--r-- 1 gus users    591 Apr 30 11:23 dna_codes.hpp
-rw-r--r-- 1 gus users   5508 Apr 30 11:23 double_fifo_input.hpp
-rw-r--r-- 1 gus users   1851 Apr 30 11:23 dumper.hpp
-rw-r--r-- 1 gus users   3976 Apr 30 11:23 eLut.h
-rw-r--r-- 1 gus users   4137 Apr 30 11:23 err.hpp
-rw-r--r-- 1 gus users   4529 Apr 30 11:23 fastq_dumper.hpp
-rw-r--r-- 1 gus users   2727 Apr 30 11:23 file_parser.hpp
-rw-r--r-- 1 gus users   1923 Apr 30 11:23 floats.hpp
-rw-r--r-- 1 gus users  17712 Apr 30 11:23 half.h
-rw-r--r-- 1 gus users    839 Apr 30 11:23 hash_function.hpp
-rw-r--r-- 1 gus users   8037 Apr 30 11:23 hash.hpp
-rw-r--r-- 1 gus users   3531 Apr 30 11:23 heap.hpp
-rw-r--r-- 1 gus users  36828 Apr 30 11:23 invertible_hash_array.hpp
-rw-r--r-- 1 gus users   9558 Apr 30 11:23 locking_hash_counters.hpp
-rw-r--r-- 1 gus users   4122 Apr 30 11:23 locks_pthread.hpp
-rw-r--r-- 1 gus users   4901 Apr 30 11:23 mapped_file.hpp
-rw-r--r-- 1 gus users   3017 Apr 30 11:23 mer_counting.hpp
-rw-r--r-- 1 gus users   4456 Apr 30 11:23 misc.hpp
-rw-r--r-- 1 gus users   9021 Apr 30 11:23 offsets_key_value.hpp
-rw-r--r-- 1 gus users   6567 Apr 30 11:23 parse_dna.hpp
-rw-r--r-- 1 gus users   5107 Apr 30 11:23 parse_quake.hpp
-rw-r--r-- 1 gus users   4806 Apr 30 11:23 parse_qual_dna.hpp
-rw-r--r-- 1 gus users   3095 Apr 30 11:23 parse_read.hpp
-rw-r--r-- 1 gus users   9631 Apr 30 11:23 raw_dumper.hpp
-rw-r--r-- 1 gus users   2354 Apr 30 11:23 read_parser.hpp
-rw-r--r-- 1 gus users    832 Apr 30 11:23 reversible_hash_function.hpp
-rw-r--r-- 1 gus users   2181 Apr 30 11:23 seq_qual_parser.hpp
-rw-r--r-- 1 gus users   2292 Apr 30 11:23 sequence_parser.hpp
-rw-r--r-- 1 gus users   3985 Apr 30 11:23 simple_circular_buffer.hpp
-rw-r--r-- 1 gus users   1874 Apr 30 11:23 simple_growing_array.hpp
-rw-r--r-- 1 gus users   5884 Apr 30 11:23 sorted_dumper.hpp
-rw-r--r-- 1 gus users   5164 Apr 30 11:23 square_binary_matrix.hpp
-rw-r--r-- 1 gus users   1057 Apr 30 11:23 storage.hpp
-rw-r--r-- 1 gus users   1404 Apr 30 11:23 thread_exec.hpp
-rw-r--r-- 1 gus users   2343 Apr 30 11:23 time.hpp
-rw-r--r-- 1 gus users 999496 Apr 30 11:23 toFloat.h
-rw-r--r-- 1 gus users   1975 Apr 30 11:23 token_ring.hpp

inst/include/jellyfish-2.2.10:
total 4
drwxr-xr-x 2 gus users 4096 Apr 30 11:24 jellyfish

inst/include/jellyfish-2.2.10/jellyfish:
total 440
-rw-r--r-- 1 gus users  2055 Apr 30 11:24 allocators_mmap.hpp
-rw-r--r-- 1 gus users  6429 Apr 30 11:24 atomic_bits_array.hpp
-rw-r--r-- 1 gus users  5229 Apr 30 11:24 atomic_field.hpp
-rw-r--r-- 1 gus users  1851 Apr 30 11:24 atomic_gcc.hpp
-rw-r--r-- 1 gus users   731 Apr 30 11:24 backtrace.hpp
-rw-r--r-- 1 gus users  7288 Apr 30 11:24 binary_dumper.hpp
-rw-r--r-- 1 gus users  3939 Apr 30 11:24 bloom_common.hpp
-rw-r--r-- 1 gus users  7237 Apr 30 11:24 bloom_counter2.hpp
-rw-r--r-- 1 gus users  5906 Apr 30 11:24 bloom_filter.hpp
-rw-r--r-- 1 gus users  5670 Apr 30 11:24 circular_buffer.hpp
-rw-r--r-- 1 gus users  2851 Apr 30 11:24 compare_and_swap.hpp
-rw-r--r-- 1 gus users  8953 Apr 30 11:24 cooperative_pool2.hpp
-rw-r--r-- 1 gus users  7931 Apr 30 11:24 cooperative_pool.hpp
-rw-r--r-- 1 gus users  4584 Apr 30 11:24 cpp_array.hpp
-rw-r--r-- 1 gus users  3201 Apr 30 11:24 divisor.hpp
-rw-r--r-- 1 gus users  3169 Apr 30 11:24 dumper.hpp
-rw-r--r-- 1 gus users  2716 Apr 30 11:24 err.hpp
-rw-r--r-- 1 gus users  4492 Apr 30 11:24 file_header.hpp
-rw-r--r-- 1 gus users  5096 Apr 30 11:24 generator_manager.hpp
-rw-r--r-- 1 gus users  6814 Apr 30 11:24 generic_file_header.hpp
-rw-r--r-- 1 gus users  8782 Apr 30 11:24 hash_counter.hpp
-rw-r--r-- 1 gus users  7484 Apr 30 11:24 int128.hpp
-rw-r--r-- 1 gus users  1577 Apr 30 11:24 jellyfish.hpp
-rw-r--r-- 1 gus users 61697 Apr 30 11:24 json.h
-rw-r--r-- 1 gus users 40441 Apr 30 11:24 large_hash_array.hpp
-rw-r--r-- 1 gus users  8023 Apr 30 11:24 large_hash_iterator.hpp
-rw-r--r-- 1 gus users  4787 Apr 30 11:24 locks_pthread.hpp
-rw-r--r-- 1 gus users  5640 Apr 30 11:24 mapped_file.hpp
-rw-r--r-- 1 gus users  1755 Apr 30 11:24 mer_dna_bloom_counter.hpp
-rw-r--r-- 1 gus users 24298 Apr 30 11:24 mer_dna.hpp
-rw-r--r-- 1 gus users  3535 Apr 30 11:24 mer_heap.hpp
-rw-r--r-- 1 gus users  3047 Apr 30 11:24 mer_iterator.hpp
-rw-r--r-- 1 gus users  9358 Apr 30 11:24 mer_overlap_sequence_parser.hpp
-rw-r--r-- 1 gus users  3706 Apr 30 11:24 mer_qual_iterator.hpp
-rw-r--r-- 1 gus users  8525 Apr 30 11:24 misc.hpp
-rw-r--r-- 1 gus users 10459 Apr 30 11:24 offsets_key_value.hpp
-rw-r--r-- 1 gus users  1419 Apr 30 11:24 parser_common.hpp
-rw-r--r-- 1 gus users 12850 Apr 30 11:24 rectangular_binary_matrix.hpp
-rw-r--r-- 1 gus users  2770 Apr 30 11:24 sam_format.hpp
-rw-r--r-- 1 gus users  4009 Apr 30 11:24 simple_circular_buffer.hpp
-rw-r--r-- 1 gus users  3845 Apr 30 11:24 sorted_dumper.hpp
-rw-r--r-- 1 gus users  5528 Apr 30 11:24 stdio_filebuf.hpp
-rw-r--r-- 1 gus users  1061 Apr 30 11:24 storage.hpp
-rw-r--r-- 1 gus users  2495 Apr 30 11:24 stream_iterator.hpp
-rw-r--r-- 1 gus users  6777 Apr 30 11:24 stream_manager.hpp
-rw-r--r-- 1 gus users  2522 Apr 30 11:24 text_dumper.hpp
-rw-r--r-- 1 gus users  1453 Apr 30 11:24 thread_exec.hpp
-rw-r--r-- 1 gus users  2343 Apr 30 11:24 time.hpp
-rw-r--r-- 1 gus users  2058 Apr 30 11:24 token_ring.hpp
-rw-r--r-- 1 gus users  6537 Apr 30 11:24 whole_sequence_parser.hpp

inst/lib:
total 7128
-rw-r--r-- 1 gus users 2065588 Apr 30 11:23 libjellyfish-1.1.a
-rwxr-xr-x 1 gus users    1058 Apr 30 11:23 libjellyfish-1.1.la
lrwxrwxrwx 1 gus users      25 Apr 30 11:23 libjellyfish-1.1.so -> libjellyfish-1.1.so.1.0.1
lrwxrwxrwx 1 gus users      25 Apr 30 11:23 libjellyfish-1.1.so.1 -> libjellyfish-1.1.so.1.0.1
-rwxr-xr-x 1 gus users 1323840 Apr 30 11:23 libjellyfish-1.1.so.1.0.1
-rw-r--r-- 1 gus users 2305398 Apr 30 11:24 libjellyfish-2.0.a
-rwxr-xr-x 1 gus users    1058 Apr 30 11:24 libjellyfish-2.0.la
lrwxrwxrwx 1 gus users      25 Apr 30 11:24 libjellyfish-2.0.so -> libjellyfish-2.0.so.2.0.0
lrwxrwxrwx 1 gus users      25 Apr 30 11:24 libjellyfish-2.0.so.2 -> libjellyfish-2.0.so.2.0.0
-rwxr-xr-x 1 gus users 1583720 Apr 30 11:24 libjellyfish-2.0.so.2.0.0
drwxr-xr-x 2 gus users    4096 Apr 30 11:24 pkgconfig

inst/lib/pkgconfig:
total 8
-rw-r--r-- 1 gus users 334 Apr 30 11:23 jellyfish-1.1.pc
-rw-r--r-- 1 gus users 334 Apr 30 11:24 jellyfish-2.0.pc

inst/share:
total 4
drwxr-xr-x 3 gus users 4096 Apr 30 11:23 man

inst/share/man:
total 4
drwxr-xr-x 2 gus users 4096 Apr 30 11:24 man1

inst/share/man/man1:
total 32
-rw-r--r-- 1 gus users 14799 Apr 30 11:23 jellyfish-1.1
-rw-r--r-- 1 gus users 14799 Apr 30 11:24 jellyfish-2.1
```

**Comment by gmarcais** [comment]
A new version of Jellyfish version 1 has been [released](https://github.com/gmarcais/Jellyfish/releases/tag/v1.1.12). It fixes some compilation issues.

**Comment by mmokrejs** [comment]
Well, Gentoo do not use `--program-suffix=` trick to often.

```
$ find /usr/portage/ -name \*.ebuild |  grep -H program-suffix
$
```

I think if you supported install into `/usr/include/jellyfish1/jellyfish.*.hpp` instead of `/usr/include/jellyfish-1.1.11/jellyfish/*.hpp` it would be closer to what I would expect. I agree provided both packages are registered through pkg-config this is a cosmetic change. At least I hope I showed that no ebuid (a package definition file) uses the `--program-suffix=` trick.

The changes showed in the original post are those which comitted to the Gentoo science tree. It could be my incompetence but I find it quite in sync with other packages.

Basically what I proposed was that by default it would be e.g. /usr/include/jellyfish{1,2} (without need for `--program-suffix=`).

**Comment by mmokrejs** [comment]
Sorry, I had to edit the above contribution due to a missing `xargs` call.

**Comment by mmokrejs** [comment]
BTW, the install procedure does not install the `doc/jellyfish.pdf` file.

**Comment by mmokrejs** [comment]
Anywa, the main concern is that 3rd-party application call both jellyfish'es as `jellyfish` so either you shall install just one version at a time or use `modules` or similar trickery to isolate the namespaces. That is a workaround only for me. If you officially stated jellyfish-1.x binary is `jellyfish1` and jellyfish-2.x provides jellyfish2, other could adjust to it in sync. Whether the header should be in `/usr/include/jellyfish{1,2}` or `/usr/include/jellyfish-2.2.10/jellyfish` is more cosmetic but still the former is shorter and I think cleaner. Um, admittedly, 3rd-party app makers would not need to edit their include lines for your `/usr/include/jellyfish-2.2.10/jellyfish`approach if they query pkg-config. Only calls to binary would need to be changed then.
